### ISSUE TITLE
Test Coverage

### ISSUE THREAD

**Issue by sebffischer** [mlr-org/mlr3torch#75]

**Comment by sebffischer** [comment]
Correctness / tests

[ ] Really test the torch learner class and write an autotest for the 
[ ] No exported functions without tests
We need proper testing for learners, some thoughts:	
don't need the autotest as we don't really test the learners
the network is generated correctly from the parameters
the dataloader is correct
the learner works correctly for regression AND classification
the dataset is correct
Parameters are properly tagged
[ ] Check that defaults and initial values are correctly used everywhere
device placement is working (use meta device)
optimizer, loss and callbacks can be properly set and defaults are correct.
Each parameter must either have a “required” tag or a default.
Improve expect_learner_torch()
[ ] properly test the shapes_out() method of the pipeoptorch
[ ] Meta tests for all test helpers 
[ ] Check that parameters either have “train” or “predict” tag (must also be checked for PipeOps)
[ ] test the autotest for pipeop torch
[ ] Check how the output names are generated, when outputs of non-terminal nodes are used
	in "output_map" ("output_<id>_output.<channel>")
